tests/msc3083_test.go

```diff
@@ -310,11 +315,13 @@ func TestRestrictedRoomsSpacesSummaryFederation(t *testing.T) {
 		},
 	})
 
-	// The room appears for no one at first since hs2 doesn't know about who is in ss1.
+	// The room appears for neither alice or bob initially. Although alice is in
+	// the space and should be able to access the room, hs2 doesn't know this!
 	requestAndAssertSummary(t, alice, space, []interface{}{space})
 	requestAndAssertSummary(t, bob, space, []interface{}{space})
 
-	// Charlie joins ss1 and now hs2 knows that alice is in it.
+	// charlie joins the space and now hs2 knows that alice is in the space (and
+	// can join room).
```

Comment:
My main question revolves around why hs2 needs to know alice can join the room if we're asking hs1 for the space summary. hs1 knows alice can join the room already.

My assumption is that hs1 needs to ask hs2 this question, but it'd be good to mention that.

Reply:
Ah, hs1 is asking hs2 about "room" while generating the summary. I'll add this.

Reply:
I updated the docstring which talks about the setup of space/room, I hope that's OK. The above is true for every time we attempt to summarize so that seemed like the best spot.

Reply:
Thanks! And aha, I think my mental model had Alice creating the room, not Charlie, that's why I was confused about why hs1 didn't know about the status of the room.

That realisation and the additional notes make things fairly clear now, thanks!